pkg/noun/manage.c

```diff
@@ -2165,7 +2171,51 @@ u3m_migrate(u3v_version ver_w)
 
   /* finally update the version and commit to disk */
   u3H->ver_w = ver_w;
-  /* extra assurance we haven't corrupted the loom before writing to disk */
+  assert(u3m_save_snapshot() == 0);
+}
+
+int
+u3m_save_snapshot(void)
+{
+  assert(u3_pma);
   u3a_loom_sane();
-  u3e_save();
+
+  if (pma_sync(u3_pma) == -1) {
+    fprintf(stderr, "snapshot: failed to save: %s\r\n", strerror(errno));
+    return -1;
+  }
+  return 0;
+}
+
+int
```

Comment:
usages of int here and in other places should be replaced by -- if more than 1 bit of info is needed - `c3_w` - otherwise `c3_t` or `c3_o`. Afterall, this is outside of /pma which debatably can differ in style.